src/mach.rs

```diff
@@ -628,27 +663,37 @@ fn build_relocations(artifact: &Artifact, symtab: &SymbolTable) -> Relocations {
                     (_, &Decl::CString {..}) => (false, X86_64_RELOC_SIGNED),
                     (_, &Decl::FunctionImport) => (false, X86_64_RELOC_BRANCH),
                     (_, &Decl::DataImport) => (false, X86_64_RELOC_GOT_LOAD),
+                };
+                match (symtab.offset(link.from.name), symtab.index(link.to.name)) {
+                    (Some(base_offset), Some(to_symbol_index)) => {
+                        debug!("{} offset: {}", link.to.name, base_offset + link.at);
+                        let builder = RelocationBuilder::new(to_symbol_index, base_offset + link.at, reloc);
+                        // NB: we currently associate absolute relocations with data relocations; this may prove
+                        // too fragile for future additions; needs analysis
+                        if absolute {
+                            segment.sections.get_index_mut(data_idx).unwrap().1.relocations.push(builder.absolute().create());
+                        } else {
+                            segment.sections.get_index_mut(text_idx).unwrap().1.relocations.push(builder.create());
+                        }
+                    },
+                    _ => error!("Import Relocation from {} to {} at {:#x} has a missing symbol. Dumping symtab {:?}", link.from.name, link.to.name, link.at, symtab)
                 }
             }
-            Reloc::Raw { .. } => panic!("unimplemented Reloc::Raw"),
-            Reloc::Debug { .. } => panic!("unimplemented Reloc::Debug"),
-        };
-        match (symtab.offset(link.from.name), symtab.index(link.to.name)) {
-            (Some(base_offset), Some(to_symbol_index)) => {
-                debug!("{} offset: {}", link.to.name, base_offset + link.at);
-                let builder = RelocationBuilder::new(to_symbol_index, base_offset + link.at, reloc);
-                // NB: we currently associate absolute relocations with data relocations; this may prove
-                // too fragile for future additions; needs analysis
-                if absolute {
-                    data_relocations.push(builder.absolute().create());
+            Reloc::Debug { size, .. } => {
+                if link.to.decl.is_section() {
+                    // TODO: not sure if these are needed for Mach
```

Comment:
Maybe put `unimplemented!()` here.

Reply:
That would prevent it working at all, eg this code path is being taken for the current example code.

Looking at a Mach object file, the debug sections don't have relocations for links to other debug sections (I've only looked at one though). So this TODO is more about understanding why and ensuring there are no corner cases, rather than actually adding some implementation here. I'm guessing dsymutil has deep knowledge of the DWARF data, so it can reconstruct the relocations without needing us to emit them, whereas ELF linkers need to be able to relocate the DWARF without parsing it.